### PR TITLE
Fix show all

### LINKED ISSUE #223: Followup for IPython integration
I got some feedback from the guys I implemented #217 for:

>For:
```
$ cat profiler.py
import time

def D():
    time.sleep(0.7)

def C():
    __tracebackhide__ = True
    time.sleep(0.1)
    D()

def B():
    __tracebackhide__ = True
    time.sleep(0.1)
    C()

def A():
    time.sleep(0.1)
    B()

A()

```
>Profiler output:
```
1.004 <module>  <string>:1
   1.004 run_path  runpy.py:245
      1.001 _run_module_code  runpy.py:89
         1.001 _run_code  runpy.py:63
            1.001 <module>  profiler.py:1
               1.001 A  profiler.py:16
                  0.700 D  profiler.py:3
                     0.700 sleep  None
                        [2 frames hidden]  <built-in>
                  0.301 sleep  None
                     [2 frames hidden]  <built-in>

```
>
>1) Inside `A`, we would like to see that a few frames (B&C) were hidden because of `__tracebackhide__`. We also suggest including a >way to expand hidden nodes in HTML format.
>2) I see, `sleep None`. Should it be `<built-in>` instead? [_Note: I don't see None in Jupyter notebook_].

I'm not entirely sure what of the above is bugs, intended behavior, or new feature requests. The request for expandable hidden nodes in html format seems like a reasonable new feature. I can start looking into making that happen.

As for "Inside `A`, we would like to see that a few frames (B&C) were hidden because of `__tracebackhide__`", I think what they mean is that when a frame is hidden due to the `__tracebackhide__` rule, the output should explicitly say that.

@joerick Can you please clarify exactly what the above example profiling output is currently saying about the hidden frames? I assume the first `[2 frames hidden]  <built-in>` line is refering to the `B` and `C` scopes. Is that right? In that case, the `<built-in>` label is a bug, and should say `<ipython built-in>` (or something) instead. What does the second repeat of the `[2 frames hidden]  <built-in>` refer to?

### PR DESCRIPTION
**Pull request by telamonian**
fixes #223

This PR address the hide/show bug that @joerick pointed out in #223. I'll open separate issues for the other discussed in that issue.

This turned out to be more complex than I would have thought for rendering related junk. In part this was because the `--show-all` cmd line option was initially busted. Passing `--show-all` on the cmd line will now ensure that `show_all=True` gets passed to the kwargs of the Renderer's constructor. I also fixed the behavior of the base Render's constructor when `show_all=True`. This together *appears* to fix the bug pointed out in #223. @joerick Can you please test this PR out when you have a sec and let me know if that part also seems fixed to you?

### CODE REVIEW

File: pyinstrument/processors.py [hunk 1]
```diff
@@ -22,40 +22,6 @@
 ProcessorOptions = Dict[str, Any]
 
 
-def remove_importlib(frame: Frame | None, options: ProcessorOptions) -> Frame | None:
```
[the remark this answers is not in the document]
Comment (telamonian) [reply pyinstrument/processors.py:25]: The only change I made to processors here is to alphabetize the order of the functions in the file. I did this in order make it easier to enumerate the processors while figuring out what to put in the `for` loop in `base.py`.

If you don't like alphabetic I'm fine with reverting this. But I would ask that we put the functions in `processors.py` in some rational order

File: pyinstrument/renderers/base.py [hunk 2]
```diff
@@ -68,7 +68,17 @@ def __init__(
...
+                processors.group_library_frames_processor,
+                processors.remove_importlib,
+                processors.remove_irrelevant_nodes,
+                processors.remove_tracebackhide,
+                processors.remove_unnecessary_self_time_nodes,
+
+                # always hide the outer pyinstrument cruft frames
+                # processors.remove_first_pyinstrument_frames_processor,
+            ):
+                self.processors.remove(p)
```
[the remark this answers is not in the document]
Comment (telamonian) [reply pyinstrument/renderers/base.py]: `remove_unnecessary_self_time_nodes` removed

**Comment by telamonian** [comment]
@joerick I've addressed your comments. Mostly I just made the changes you suggested.

The output looks good with `--show-all` present, but without it the placement and explanation of the hidden nodes still seems kinda iffy. I think it has something to do with the grouping logic in `remove_unnecessary_self_time_nodes` from `processors.py`. I'll keep working on that. Regardless, I think this might be a good place to wrap up this particular PR, but that's up to you.

To clarify, I'm profiling this script:

```python
import time

def D():
    time.sleep(0.7)

def C():
    __tracebackhide__ = True
    time.sleep(0.1)
    D()

def B():
    __tracebackhide__ = True
    time.sleep(0.1)
    C()

def A():
    time.sleep(0.1)
    B()

A()
```

For command:
```
pytinstrument --show-all thide.py
```

output:
```
  _     ._   __/__   _ _  _  _ _/_   Recorded: 12:28:51  Samples:  4
 /_//_/// /_\ / //_// / //_'/ //     Duration: 1.002     CPU time: 0.001
/   _/                      v4.4.0

Program: thide.py

1.002 <module>  thide.py:1
└─ 1.002 A  thide.py:16
   ├─ 0.901 B  thide.py:11
   │  ├─ 0.801 C  thide.py:6
   │  │  ├─ 0.701 D  thide.py:3
   │  │  │  └─ 0.701 sleep  None
   │  │  └─ 0.100 sleep  None
   │  └─ 0.100 sleep  None
   └─ 0.101 sleep  None

To view this report with different options, run:
    pyinstrument --load-prev 2023-01-17T12-28-51 [options]
```

For command:
```
pyinstrument thide.py
```

output:
```  _     ._   __/__   _ _  _  _ _/_   Recorded: 12:30:56  Samples:  4
 /_//_/// /_\ / //_// / //_'/ //     Duration: 1.002     CPU time: 0.001
/   _/                      v4.4.0

Program: thide.py

1.002 <module>  thide.py:1
└─ 1.002 A  thide.py:16
   ├─ 0.701 D  thide.py:3
   │  └─ 0.701 sleep  None
   │        [2 frames hidden]  <built-in>
   └─ 0.301 sleep  None
         [2 frames hidden]  <built-in>

To view this report with different options, run:
    pyinstrument --load-prev 2023-01-17T12-30-56 [options]
```

**Comment by mattip** [comment]
Hi. I am interested in pushing forward with this PR to get the feature in. @telamonian there is a linting issue. Can you invite me as a collaborator to your fork so I can push directly to the branch, or should I issue a new PR based on this one, cherry-picking your commits and adding one to fix the linting?

**Comment by joerick** [comment]
superceded by #239

**Pull request closed without merging**